src/server/game/Movement/Waypoints/WaypointManager.cpp

```diff
@@ -68,25 +52,25 @@ void WaypointMgr::Load()
         Trinity::NormalizeMapCoord(x);
         Trinity::NormalizeMapCoord(y);
 
-        wp->id = fields[1].GetUInt32();
-        wp->x = x;
-        wp->y = y;
-        wp->z = z;
-        wp->orientation = o;
-        wp->move_type = fields[6].GetUInt32();
+        WaypointData waypoint;
+        waypoint.id = fields[1].GetUInt32();
+        waypoint.x = x;
+        waypoint.y = y;
+        waypoint.z = z;
+        waypoint.orientation = o;
+        waypoint.move_type = fields[6].GetUInt32();
 
-        if (wp->move_type >= WAYPOINT_MOVE_TYPE_MAX)
+        if (waypoint.move_type >= WAYPOINT_MOVE_TYPE_MAX)
         {
-            TC_LOG_ERROR("sql.sql", "Waypoint %u in waypoint_data has invalid move_type, ignoring", wp->id);
-            delete wp;
+            TC_LOG_ERROR("sql.sql", "Waypoint %u in waypoint_data has invalid move_type, ignoring", waypoint.id);
             continue;
         }
 
-        wp->delay = fields[7].GetUInt32();
-        wp->event_id = fields[8].GetUInt32();
-        wp->event_chance = fields[9].GetInt16();
+        waypoint.delay = fields[7].GetUInt32();
+        waypoint.event_id = fields[8].GetUInt32();
+        waypoint.event_chance = fields[9].GetInt16();
 
-        path.push_back(wp);
+        path.push_back(std::move(waypoint));
```

Comment:
std::move here is totally not needed and may be harmfull. Why?  WaypointData is POD struct and calling std::move on it will only invoke copy constructor/assignment call on push_back.

Why can it be harmful? You are preventing possible copy ellision optimisation done by compiler in this case ;]